pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -40,14 +48,45 @@ func nameAndTag(id string) (name string, tag string, err error) {
 		name = segments[0]
 		tag = segments[1]
 		if len(name) == 0 || len(tag) == 0 {
-			err = errors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
+			err = kapierrors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
 		}
 	default:
-		err = errors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
+		err = kapierrors.NewBadRequest("ImageStreamTags must be retrieved with <name>:<tag>")
 	}
 	return
 }
 
+func (r *REST) List(ctx kapi.Context, label labels.Selector, field fields.Selector) (runtime.Object, error) {
+	imageStreams, err := r.imageStreamRegistry.ListImageStreams(ctx, labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	matcher := MatchImageStreamTag(label, field)
+
+	list := &api.ImageStreamTagList{}
+	for _, currIS := range imageStreams.Items {
+		for currTag := range currIS.Status.Tags {
+			uncast, err := r.Get(ctx, api.JoinImageStreamTag(currIS.Name, currTag))
```

Comment:
I expected an argument here since `Get` will retrieve the same imagestream that I already have over and over again.  Is this the N+1 you're talking about and you hit the wrong line?  The refactor isn't big.